shell/platform/windows/flutter_window.cc

```diff
@@ -74,11 +74,19 @@ FlutterWindow::FlutterWindow(int width, int height)
   current_cursor_ = ::LoadCursor(nullptr, IDC_ARROW);
 }
 
-FlutterWindow::~FlutterWindow() {}
+FlutterWindow::~FlutterWindow() {
+  OnWindowStateEvent(WindowStateEvent::kHide);
```

Comment:
Imagine I'm an add-to-app scenario where most of my app's content is built using win32 APIs, but one page is built using Flutter:

1. The user launches the app. A top-level window is created using win32 APIs
2. The user navigates to the "Home" page, which is built using win32 APIs
3. The user navigates to the "Settings" page, which is built using Flutter. The app creates a `FlutterViewController`, which creates a `FlutterWindowsView` and `FlutterWindow`.
4. The user navigates back to the "Home" win32 page. The app no longer needs the `FlutterViewController`, so it destroys it which also destroys the `FlutterWindowsView` and `FlutterWindow`.

The app's window is visible and focused at the end of this scenario. However, from my understanding this line will update the app's state to hidden.

Some questions:
1. Can we add a test that covers this scenario? It'd be easy to regress this behavior
2. Should we remove this line? We send the "hide" state event when the top-level Flutter window receives a `WM_DESTROY` - is that enough?

Reply:
Another similar concern: an add-to-app scenario where the Flutter view's `HWND` has width & height `0` but the top-level window has non-zero size. This should not result in a hidden lifecycle event.

Reply:
I do not follow your logic in the first concern. It will eventually call `WindowsLifecycleManager::OnWindowStateEvent`, which will only update the state to hidden when there are no other currently visible windows (`visible_windows_.empty()`).

Regarding the second, do you propose we should not respond to the flutter window's `WM_SIZE` messages? That could work, but the developer would need to make sure the API function is called if they want top-level window focus to effect the lifecycle state, because, again, the top-level window under normal circumstances receives `GETFOCUS` messages upon initialization with no paired `KILLFOCUS` when the flutter app loses focus, so we cannot listen for them in the top-level message handler.

Reply:
Sorry, I forgot that both the top-level and inner `HWND`s have separate window states.

> Regarding the second, do you propose we should not respond to the flutter window's `WM_SIZE` messages?

I meant to say that the app should not receive a `AppLifecycleState.hidden`. Sending a "hidden" window state event here seems reasonable. It seems your code covers this concern: the app won't receive a "hidden" state as the top-level window remains visible.

Would you mind adding some tests to cover these scenarios though? Perhaps:

1. There's an engine and view. Both the top-level and inner `HWND`s are visible. The app remains in the "resumed" state if the view is destroyed.
2. There's an engine and view. Both the top-level and inner `HWND`s are visible. The app remains in the "resumed" state if the view is resized to width & height `0`.

Feel free to do whichever variants you think would be most valuable :)

Reply:
When you say "the view is destroyed", does this mean the `FlutterWindowsView` is destructed? Because, if so, doing so stops the engine, so platform messages to the framework will cease. Also, in either case, should the app stay in resumed, or inactive, if the inner HWND was the only focused one before being hidden? I would think inactive, rather than resumed.